numpy/compat/_inspect.py

```diff
@@ -184,9 +184,8 @@ def formatargvalues(args, varargs, varkw, locals,
     def convert(name, locals=locals,
                 formatarg=formatarg, formatvalue=formatvalue):
         return formatarg(name) + formatvalue(locals[name])
-    specs = []
-    for i in range(len(args)):
-        specs.append(strseq(args[i], convert, join))
+    specs = [strseq(args[i], convert, join) for i in range(len(args))]
```

Comment:
Maybe I'm missing something, but do you really need the index here? Why not:
```
specs = [strseq(arg, ...) for arg in args]
```

Reply:
You are right -- simplified it more.